.storybook/docgen-plugin.js

```diff
@@ -1,5 +1,7 @@
 const docGen = require('react-docgen-typescript');
-const docGenLoader = require('react-docgen-typescript-loader/dist/generateDocgenCodeBlock.js');
+const {
+  generateDocgenCodeBlock,
+} = require('@storybook/react-docgen-typescript-plugin/dist/generateDocgenCodeBlock.js');
```

Comment:
`react-docgen-typescript-loader` doesn't seem to be maintained and does use the newest version of `react-docgen-typescript`. The project was forked for use in Storybook.

Reply:
This import was weird but I looked and saw that `generateDocgenCodeBlock` wasn't exported from the index. I'm not sure if the `.js` extension is needed but maybe because of how it wasn't exported publicly.

Reply:
That's a good question. `generateDocgenCodeBlock` wasn't exported from `react-docgen-typescript-loader`, but I'll check `@storybook/react-docgen-typescript-plugin`

Reply:
`generateDocgenCodeBlock` is not exported: https://github.com/hipstersmoothie/react-docgen-typescript-plugin/blob/master/src/index.ts

We make our own plugin for processing prop types because we do extra processing. The Storybook plugin does not allow injecting any options to the generator.

For example:
- we mark the `as` prop as optional because `react-docgen-typescript` thinks it is supposed to be required because of the overload.
- We rewrite `as` prop type to a list of valid elements instead of `React.Element<any>`
- We filter style props from `Box` and instead link to `Box`. If we didn't do this, the prop list would be huge and the same for almost all compound components